### DISCUSSION TITLE
NVME support status (write speed problem)

### DISCUSSION THREAD

**Discussion by motorics** [Joshua-Riek/ubuntu-rockchip#45] [Q&A]
Hi,

First of all I want to say a big big thank you for your work!!!
I would like to ask you about the NVME support status. Should it work fine?

I am asking because I am facing a strange issue... I have installed your release on the NVME and while if I am downloading something from the internet directly on the NVME (http://speed.hetzner.de/10GB.bin for example) I am experiencing a very very low speed (4-8 MegaBytes/s in browser, 55MB/s dropped suddely after 300 MB downloaded to 4-7MB/s and after couple of seconds it will jump back to 55MB/s and decrease again to <7MB/s) while if I copy the same file to the attached external HDD (Seagate Basic 4TB into the USB3) it will copy it with 48-55 MB/s. 

By testing the same by redirecting the output of wget into the /dev/null I will obtain still the 44-55 MB/s constantly, also I had run iperf3 with public server and inside my network in order to stress the network and everything was fine and stable. So we can assume that the Gigabit  ethernet module is working fine.

Also another strange behavior is that applications like qbittorrent-nox installed on the NVME but saving the files on the external harddrive are not able to keep the download speed they will start to download the torrent will increase to 45-55 MB/s and suddenly will decrease to couple of kilobytes and struggle for 20 seconds to regain the full download speed for just couple of seconds.
I have tested the NVME in another SBC (UP Squared) and it is working fine. Also on a powerfull Windows computer by using a benchmark it will be able to read and write with 1.2GB/s and the health is 100%, basically it is brand new, 1 month old.
By copying a file from the attached external drive on the NVME (big file, 25GB) it will behave normally and will keep a copy speed between 118-122MB/s (the vice-versa from nvme to hdd is still working fine and stable)

All the tests I have repeated for at least 5-6 times before write my concern and I have restarted the device many times also by booting the clean image, also by running the upgrade.

Many thanks in advance for your feedback ! and if I the issue still persists on your side too, please let me know how can I help you in the TShooting.

**Comment by Joshua-Riek** [comment]
NVMe should be working fine. Does your hard drive have an external power source? Or is it only powered through USB? I will do some tests on my end as well.

**Comment by motorics** [comment]
Yes, I has tried by connected it into a USB 3.0 hub with power supply also.
But I had run also tests with the harddrive disconnected from the OrangePi
just to make sure that it will not interfere somehow... Still same results.

Thank you !

**Reply by ferrellsl** [reply]
I recommend running a disk benchmarking tool such as KDiskMark.  I have a 256GB NVME that  I purchased on AliExpress.  Can't even remember the brand as it wasn't important, but here are my results.

![kdiskmark results](https://user-images.githubusercontent.com/5926287/222937152-aa6cf29e-914e-40c7-808a-9296bd92dc1d.png)

**Comment by Joshua-Riek** [comment]
Here are the results on my system. I see the graph drops to about 100 MB/s near the end of the test. But it should not be noticed in typical usage.
![Screenshot from 2023-03-04 21-44-46](https://user-images.githubusercontent.com/10427125/222939068-eb6c04ff-7e07-435e-a708-d00ba7cc3bbf.png)

**Reply by motorics** [reply]
By booting from an SD card and performing the same benchmark by selecting the nvme I obtain the same results like you (405MB/s read | 395 MB/s write).
I had tried by reinstalling the release on the NVME but issue still persists.

Could you please confirm if I am installing it in the correct way?

Basically I have write your image on an SD card with Raspberry Pi Imager.
Boot from SDCard
Download the same image on the SD card.
Delete all partitions from NVME by using cfdisk.
And use the dd if=ubuntu-22.04-preinstalled-desktop-arm64-orange-pi5.img of=/dev/nvme0n1 status=progress
because I did not find the SPI flasher image on the SD card I have clone the git and compile it (on another machine) and took from there the SPI image and using DD I write it.
turn off, remove microsd card and turn on.

Many thanks in advance !
p.s. I see that you have released version 1.8 I will try it now :)

**Reply by Joshua-Riek** [reply]
The SPI image is now in the latest release. You are installing it correctly. If you perform the same disk benchmark when booting from the NVME do you get the same results?

**Reply by motorics** [reply]
Hello,
I can't perform benchmarking while I am booting from NVME just read benchmark which is good (>400 MB/s), write benchmark is not possible because the benchmark will try to unmount the drive and I will receive an error pop-up :Error unmounting filesystem: target is busy. I will try now with KDiskMark

**Reply by motorics** [reply]
![Screenshot from 2023-03-05 19-05-11](https://user-images.githubusercontent.com/9324056/222981272-149a799f-e2b0-43e9-a9bc-f6de18239f0b.png)
![Screenshot from 2023-03-05 19-15-37](https://user-images.githubusercontent.com/9324056/222981275-fc18a93b-3a5f-4e3b-8ee5-c741f2b37603.png)
